includes/REST_API/Media_Hotlinking_Controller.php

```diff
@@ -0,0 +1,290 @@
+				$data[ $check_field ] = rest_sanitize_value_from_schema( $link[ $check_field ], $schema['properties'][ $check_field ] );
+			}
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+
+		return $response;
+	}
+
+	/**
+	 * Retrieves the link's schema, conforming to JSON Schema.
+	 *
+	 * @since 1.11.0
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema(): array {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$schema = [
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'link',
+			'type'       => 'object',
+			'properties' => [
+				'ext'       => [
+					'description' => __( 'File extension', 'web-stories' ),
+					'type'        => 'string',
+					'context'     => [ 'view', 'edit', 'embed' ],
+				],
+				'file_name' => [
+					'description' => __( 'File name', 'web-stories' ),
+					'type'        => 'string',
+					'context'     => [ 'view', 'edit', 'embed' ],
+				],
+				'file_size' => [
+					'description' => __( 'File size', 'web-stories' ),
+					'type'        => 'integer',
+					'context'     => [ 'view', 'edit', 'embed' ],
+				],
+				'mime_type' => [
+					'description' => __( 'Mime Type', 'web-stories' ),
+					'type'        => 'string',
+					'context'     => [ 'view', 'edit', 'embed' ],
+				],
+			],
+		];
+
+		$this->schema = $schema;
+
+		return $this->add_additional_fields_schema( $this->schema );
+	}
+
+	/**
+	 * Checks if current user can process links.
+	 *
+	 * @since 1.11.0
+	 *
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function parse_url_permissions_check() {
+		if ( ! $this->get_post_type_cap( Story_Post_Type::POST_TYPE_SLUG, 'edit_posts' ) ) {
+			return new WP_Error( 'rest_forbidden', __( 'Sorry, you are not allowed to insert external media.', 'web-stories' ), [ 'status' => rest_authorization_required_code() ] );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Callback to validate urls.
+	 *
+	 * @since 1.11.0
+	 *
+	 * @param mixed $value Value to be validated.
+	 *
+	 * @return true|WP_Error
+	 */
+	public function validate_url( $value ) {
+		$url = untrailingslashit( $value );
+
+		if ( empty( $url ) ) {
+			return new WP_Error( 'rest_invalid_url', __( 'Invalid URL', 'web-stories' ), [ 'status' => 400 ] );
+		}
+
+		$parts = wp_parse_url( $url );
+
+		if ( ! isset( $parts['host'] ) ) {
+			return new WP_Error( 'rest_invalid_url_host', __( 'Invalid URL Host', 'web-stories' ), [ 'status' => 400 ] );
+		}
+
+		if ( ! isset( $parts['path'] ) || empty( $parts['path'] ) ) {
+			return new WP_Error( 'rest_invalid_url_path', __( 'Invalid URL Path', 'web-stories' ), [ 'status' => 400 ] );
+		}
```

Comment:
Can you elaborate on why this was added? It seems overly defensive. If I host a random cat image at `https://pascalscatimage.com`, this would be blocked by this check, no?

Reply:
The user can submit whatever they want to this url. 

So first case. 

#### Random string - example wibble. 
This would be regrets, as it would fail to be parsed as a url

#### Relative path /image/test.png
User could reasonably think that they can use a relative path. I think we should avoid this, as it is messy. So would not have a host, which is why we should regret it. 

#### Top level domain - https://pascalscatimage.com
Why anyone do this, it is insane. Either way, the likelyhood is low that this is valid. It also generates other problems. Our code requires a file, to say generate a poster image etc. Not having a file name would break a lot of stuff. I think it is an edge can we should not allow.